sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/ValidationsTest.java

```diff
@@ -0,0 +1,152 @@
+        final File[] filteredFiles = folderFile.listFiles(pathname -> pathname.getName().toLowerCase().contains(fileNameFilter));
+        assert filteredFiles != null;
+
+        Arrays.sort(filteredFiles, Comparator.comparing(File::getName));
+        return filteredFiles;
+    }
+
+    private List<ValidationTestCase> readTestcasesFromFile(File testFile) throws IOException {
+        final String jsonString = Files.readString(testFile.toPath());
+        final CollectionType typeReference =
+            TypeFactory.defaultInstance().constructCollectionType(List.class, ValidationTestCase.class);
+        return objectMapper.readValue(jsonString, typeReference);
+    }
+
+    @SuppressWarnings("unchecked")
+    private LinkedHashMap<String, Object> readConfigurationFromFile(File sampleFile) throws IOException {
+        final String jsonString = Files.readString(sampleFile.toPath());
+        final LinkedHashMap<String, Object> configurations = objectMapper.readValue(jsonString, new TypeReference<>() {
+        });
+        final Object featureManagementSection = configurations.get("feature_management");
+        if (featureManagementSection.getClass().isAssignableFrom(LinkedHashMap.class)) {
+            return (LinkedHashMap<String, Object>) featureManagementSection;
+        }
+        return new LinkedHashMap<>();
+    }
+
+    @SuppressWarnings("unchecked")
+    private void runTestcases(File sampleFile, File testsFile) throws IOException {
+        // initialize feature manager
+        final FeatureManagementProperties managementProperties = new FeatureManagementProperties();
+        managementProperties.putAll(readConfigurationFromFile(sampleFile));
+        final FeatureManager featureManager = new FeatureManager(context, managementProperties, configProperties);
+
+        final List<ValidationTestCase> testCases = readTestcasesFromFile(testsFile);
+        System.out.println("Running test case from file: " + testsFile.getName());
+
+        for (int i = 0; i < testCases.size(); i++) {
+            System.out.println("Test case " + i + " : " + testCases.get(i).getDescription());
+            if (hasException(testCases.get(i))) {   // TODO. Currently we didn't throw the exception when parameter is invalid
+                assertNull(managementProperties.getOnOff().get(testCases.get(i).getFeatureFlagName()));
+            } else {
+                if (hasInput(testCases.get(i))) { // Set inputs
+                    final Object userObj = testCases.get(i).getInputs().get(inputsUser);
+                    final Object groupsObj = testCases.get(i).getInputs().get(inputsGroups);
+                    final String user = userObj != null ? userObj.toString() : null;
+                    final List<String> groups = groupsObj != null ? (List<String>) groupsObj : null;
+                    when(context.getBean(Mockito.contains("Targeting"))).thenReturn(new TargetingFilter(new TargetingFilterTestContextAccessor(user, groups)));
```

Comment:
This should be removed. It currently isn't being used, and shouldn't be needed once this is fixed.

Reply:
It's used when running test case from "TargetingFilter.modified.test.json". The test case from modified file can succeeded. 
Shall we also ignore the test case from "TargetingFilter.modified.test.json" even if it succeeded now?

Reply:
I think this issue is that it just happens to work and if we start adding other tests once we get this all done it could break randomly.

Reply:
Ignore both "TargetingFilter.modified.test.json" and "TargetingFilter.test.json"